utils/ioutil.go

```diff
@@ -46,10 +46,26 @@ type IoUtil interface {
 	GetPluginMetas(plugins []model.PluginFQN, defaultRegistry string) ([]model.PluginMeta, error)
 }
 
-type impl struct{}
+type HttpClient interface {
+	Get(URL string) (resp *http.Response, err error)
+}
+
+type impl struct {
+	httpClient HttpClient
+}
 
+// New creates an instance of IoUtil using the default http client.
 func New() IoUtil {
-	return &impl{}
+	return &impl{
+		httpClient: http.DefaultClient,
+	}
+}
+
+// NewWithClient creates an instance of IoUtil using the provided client for all requests.
+func NewWithClient(client HttpClient) IoUtil {
```

Comment:
Can you elaborate why do we need this function?

Reply:
This method is used to mock HTTPClient in the IoUtil tests; it makes it easier to test request failure, etc and makes the tests more robust. See utils/mocks/MockHttpClient.go

Reply:
There is a simpler mocking technique. See explanation how it is used in VS Code broker tests in one of my other comments